lib/estella/searchable.rb

```diff
@@ -57,10 +57,10 @@ def as_indexed_json(_options = {})
     end
 
     module ClassMethods
-      # support for mongoid::slug
+      # support for mongoid::slug and friendly_id (AR slugs)
       # indexes slug attribute by default
       def index_slug
-        if defined? slug
+        if method_defined?(:slug) || method_defined?(:friendly_id)
```

Comment:
This seems OK, if it works equivalently in practice. (We saw that it might depend on load order.)

Since friendly_id also [offers a class-level `friendly_id` helper](https://github.com/norman/friendly_id?tab=readme-ov-file#usage), if you wanted to stick with the old style, I suspect you could alternatively do `defined? slug || defined? friendly_id`.

Reply:
I ended up choosing to use `method_defined?` because of consistent and predictable behavior when compared to using `defined?` with a logical `||`, which still confuses me and I think it breaks the condition. Let me explain with an example.

Initializing `User`, a mongoid model that includes `Searchable` but does not define slug method or include `Mongoid::Slug`:
```ruby
[1] pry(User)> defined? slug
=> nil
[2] pry(User)> defined? slug || defined? friendly_id
=> "expression"
[3] pry(User)> method_defined?(:slug) || method_defined?(:friendly_id)
=> false
```

Then Initializing `Artist`, a model that includes `Searchable` module and `Mongoid::Slug`:
```ruby
[1] pry(Artist)> defined? slug
=> "method"
[2] pry(Artist)> defined? slug || defined? friendly_id
=> "expression"
[3] pry(Artist)> method_defined?(:slug) || method_defined?(:friendly_id)
=> true
```

Now Sale, an AR model that incudes `Searchable` and `FriendlyId` modules:
```ruby
[1] pry(Sale)> defined? slug
=> nil
[2] pry(Sale)> defined? slug || defined? friendly_id
=> "expression"
[3] pry(Sale)> method_defined?(:slug) || method_defined?(:friendly_id)
=> true
```

So by my understanding, using `defined? slug || defined? friendly_id` is going to evaluate for models even if they don't include (or implement) `slug` or `friendly_id` methods.

I've tested the initialization and `method_defined?` seems to work for this. The load order IIRC was related to when we used `slugs` method. But for `friendly_id`, it seems to be loaded in time, before initializing the models.

Reply:
Wow OK. I think this strange result:
```ruby
defined? slug || defined? friendly_id # expression
```
...has to do with the precedence of the `defined?` operator (which is _not_ a regular ruby method). I think it was doing passing `slug || defined? friendly_id` to that initial `defined?`, and parentheses would help make it more consistent.

However, this just demonstrates how `defined?` can be confusing so your solution is probably best.